codeguru_profiler_agent/aws_lambda/lambda_handler.py

```diff
@@ -39,10 +29,60 @@ def load_handler(bootstrap_module, env=os.environ, original_handler_env_key=HAND
         raise
 
 
-# Load the customer's handler, this should be done at import time which means it is done when lambda frameworks
-# loads our module. We load the bootstrap module by string name so that IDE does not complain
-lambda_bootstrap_module = __import__("bootstrap")
-handler_function = load_handler(lambda_bootstrap_module)
+def _python36_extractor(bootstrap_module, original_handler_name):
+    """
+    The lambda bootstrap code for python 3.6 was different than for later versions, instead of the _get_handler
+    function there was a more complex _get_handlers function with more parameters
+    """
+    # TODO FIXME Review if the support for python 3.6 bootstrap can be improved.
+    # This returns both a init_handler and the function, we apply the init right away as we are in init process
+    init_handler, customer_handler_function = bootstrap_module._get_handlers(
+        handler=original_handler_name,
+        mode='event',  # with 'event' it will return the function as is (handlerfn in the lambda code)
+        # 'http' would return wsgi.handle_one(sockfd, ('localhost', 80), handlerfn) instead
+        invokeid='unknown_id')  # FIXME invokeid is used for error handling, need to see if we can get it
+    init_handler()
+    return customer_handler_function
+
+
+def get_lambda_handler_extractor():
+    """
+    This loads and returns a function from lambda or RIC source code that is able to load the customer's
+    handler function.
+    WARNING !! This is a bit dangerous since we are calling internal functions from other modules that we do not
+    officially depend on. The idea is that this code should run only in a lambda function environment where we can know
+    what is available. However if lambda developers decide to change their internal code it could impact this !
+    """
+    # First try to load the lambda RIC if it is available (i.e. python 3.9)
+    # See https://github.com/aws/aws-lambda-python-runtime-interface-client
+    ric_bootstrap_module = _try_to_load_module("awslambdaric.bootstrap")
+    if ric_bootstrap_module is not None and hasattr(ric_bootstrap_module, '_get_handler'):
+        return ric_bootstrap_module._get_handler
+
+    # If no RIC module is available there should be a bootstrap module available
+    # do not catch ModuleNotFoundError exceptions here as we cannot do anything if this fails.
```

Comment:
I don't understand this comment and its place here. Should this be moved in the `_try_to_load_module` method?

Reply:
basically I first try to load module `awslambdaric.bootstrap` and if that is not available I try to load `bootstrap`.
While the first one is never available for lambdas with runtime < 3.9 (at the moment), the other should always be there. So I use `_try_to_load_module` to get the first one but for the second I do `import_module` directly, not trying to catch any exception if it fails.